packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -252,40 +275,36 @@ export class TokenDetectionController extends PollingControllerV1<
   async detectTokens(options?: {
     networkClientId?: NetworkClientId;
     accountAddress?: string;
-  }) {
-    const { networkClientId, accountAddress } = options || {};
-    const {
-      disabled,
-      isDetectionEnabledForNetwork,
-      isDetectionEnabledFromPreferences,
-    } = this.config;
+  }): Promise<void> {
+    const { networkClientId, accountAddress } = options ?? {};
     if (
-      disabled ||
-      !isDetectionEnabledForNetwork ||
-      !isDetectionEnabledFromPreferences
+      this.#disabled ||
+      !this.#isDetectionEnabledForNetwork ||
+      !this.#isDetectionEnabledFromPreferences
     ) {
       return;
     }
-    const { tokens } = this.getTokensState();
-    const selectedAddress = accountAddress || this.config.selectedAddress;
-    const chainId = this.getCorrectChainId(networkClientId);
-
-    const tokensAddresses = tokens.map(
+    const { tokens } = this.#getTokensState() ?? {};
+    const selectedAddress = accountAddress ?? this.#selectedAddress;
+    const chainId = this.#getCorrectChainId(networkClientId);
+    const tokensAddresses = tokens?.map(
       /* istanbul ignore next*/ (token) => token.address.toLowerCase(),
     );
-    const { tokenList } = this.getTokenListState();
+    const { tokenList } = await this.messagingSystem.call(
```

Comment:
Is this `async`?

Reply:
Ah good catch! fixed here: https://github.com/MetaMask/core/pull/3609/commits/2bef9b81b573db9b459f26b544b3c67824ed233e